Updated German localization

claudiush opened the pull request:
- More consistent glossary use
- Corrected punctuation
- Removed "Einkaufenladen", which is not a common German word

deathbaba commented on data/categories.txt:
```diff
@@ -1249,7 +1249,7 @@ amenity-place_of_worship
 ru:3храм|церковь|достопримечательность
 uk:3храм|церква|собор|пам’ятка|алтарь|мавзолей|пам’ятні місця
-de:4Tempel|Kirche|Dom|Sehenswürdigkeit
+de:Kirche|Synagoge|Moschee|4Tempel|Anbetungsstätte|Dom|Sehenswürdigkeit
```
This translation should have only generic words, not related to any specific religion.
We have recently introduced religion-specific types but still not translated them all at the moment.

deathbaba commented on data/categories.txt:
```diff
@@ -1510,6 +1510,7 @@ amenity-bench
 en:bench
 ru:скамейка|лавка
 uk:лавка
+en:Parkbank
```
de: instead of en:

deathbaba commented on data/categories.txt:
```diff
@@ -2814,7 +2815,7 @@ highway-primary|highway-primary_link|highway-residential|highway-secondary|highw
 ru:улица|переулок|проезд|тракт|проспект|ул
 uk:вулиця|провулок|проїзд|проспект|бульвар|вул
-de:Strasse|Weg|Avenue|Gasse|str
+de:Strasse|Weg|Avenue|Gasse|Str
```
@vng @ygorshenin @mpimenov Does it fit well with our search_string_utils values? Should we store all synonyms in one place only instead of two?

ygorshenin replied:
There is some kind of duplication, you're right. Seems that we should parse somehow this file to create a list of synonyms and use it during search.

claudiush commented:
Thanks for the review. I have incorporated the fixes you mentioned and also added the new separate category entries for religion specific place of worship POIs.
